### ISSUE TITLE
Max Pooling Ordering/Feature Maps

### ISSUE THREAD

**Issue by pGit1** [EricAlcaide/SimpleNet-Keras#1]
Why do you switch order of MaxPooling operator sometimes and sometimes not? I don't gather this from the paper.

```
    # Block 4
    model.add(Conv2D(128, (3,3), padding='same', kernel_initializer=RandomNormal(stddev=0.01)))
    model.add(BatchNormalization())
    model.add(Activation(act))
    # First Maxpooling
    model.add(MaxPooling2D(pool_size=(2,2), strides=s))
    model.add(Dropout(0.2))

    # Block 7
    model.add(Conv2D(256, (3,3), padding='same', kernel_initializer=glorot_normal()))
    # Second Maxpooling
    model.add(MaxPooling2D(pool_size=(2,2), strides=s))
    model.add(BatchNormalization())
    model.add(Activation(act))
    model.add(Dropout(0.2))

```

You add maxpool after batch norm and activation in one instance and before in another. What is intuition for doing this? I don't find this in any of the paper.

I notice you also grow the number of kernels as network gets deeper but its sporadic. The figure in the paper seems that it keeps the number fixed at 128 but it advocates "cone shaped" networks with increasing feature map sizes which leads to my confusion.

**Comment by pGit1** [comment]
Beginning to think there is no true reason for this.

**Comment by Coderx7** [comment]
@pGit1 : 
@EricAlcaide is right. the main architecture is not any different. 
Those changes are made based on the last principle, i.e to decrease memory usage.(find the best compromise on accuracy/performance for ones requirements).  
The reason MaxPooling was used before BatchNormalization was solely to reduce memory usage. Caffe was very bad (and still is compared to other frameworks such as torch/TF) at memory usage.  
The idea in that paper was to first create the architecture, and then modify it to have the best compromise for your situation.  
If you want to have good intuitions concerning how to develop an architecture or resolve any confusions created so far, I strongly suggest you to have a look at here : https://github.com/Coderx7/SimpNet 
This is actually a follow up work, which was done to address many questions raised form SimpleNet paper. 
In this paper, I tried to thoroughly explain each and every principles and also improve the overall work as well. the result is the improved version of the architecture which is better than the SimpleNet and while having the same number of parameters, it outperforms deeper and more complex architectures such as WRN easily. ( As a quick note, SimpleNet could reach 73.x% on CIFAR10 while SimpNet reaches up to 79.x%, )
if you happen to have any follow up questions please email me or create an issue on the corresponding repository (though emailing is preferred)
